[[ Bug 22999 ]] Fix background location updates on iOS

livecodepanos opened the pull request:
This patch ensures that the app can still receive location updates when suspended, if the "location" is among the chosen UIBackgroundModes.

livecodeian commented on engine/src/mblhandlers.cpp:
```diff
@@ -1336,6 +1336,31 @@ Exec_stat MCHandleSensorAvailable(void *p_context, MCParameter *p_parameters)
+        MCAutoValueRef t_value;
+        MCAutoBooleanRef t_bool;
+        p_parameters->eval_argument(ctxt, &t_value);
+        // PM-2015-03-11: [[ Bug 14855 ]] Evaluate correctly the second param
```
I don't think this comment belongs here (copy & pasted?)

livecodepanos replied:
Oops  :)

livecodeian commented on engine/src/mbliphonesensor.mm:
```diff
@@ -294,6 +294,25 @@ static void initialize_core_location(void)
+    if (t_plist_can_allow_background_location_updates)
+    {
+        if (p_allow)
+            s_location_manager.allowsBackgroundLocationUpdates = YES;
+        else
+            s_location_manager.allowsBackgroundLocationUpdates = NO;
```
@livecodepanos This function needs a call to `initialize_core_location` before accessing members of `s_location_manager`

livecodepanos replied:
@livecodeian done :)

livecodepanos committed:
[[ Bug 22999 ]] Allow background location updates on iOS

This patch adds a new command "iphoneAllowBackgroundLocationUpdates" that
can allow/disallow location updates when the app is suspended.

livecodeian commented:
@livecode-vulcan review ok 704308d

livecode-vulcan commented:
💙 review by @livecodeian ok 704308debb03927c9845d0a6d9abab358fbe795e

livecode-vulcan commented:
😎 test success 704308debb03927c9845d0a6d9abab358fbe795e

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1021)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/429)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/428)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/427)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/428)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2546)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/23)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/75)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/23)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/74)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2572)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2562)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2560)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2597)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2533)
